src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -589,7 +589,7 @@ protected function getAbsoluteUri($uri)
             $uri = $path.$uri;
         }
 
-        return preg_replace('#^(.*?//[^/]+)\/.*$#', '$1', $currentUri).$uri;
+        return preg_replace('#^([^/]*//[^/]+)\/.*$#', '$1', $currentUri).$uri;
```

Comment:
Those are not equivalent. You cannot replace `.*?` or `.+?` like you are doing. All those changes should be reverted.